redis-om-spring/src/main/java/com/redis/om/spring/metamodel/MetamodelGenerator.java

```diff
@@ -244,6 +249,8 @@ private List<Triple<ObjectGraphFieldSpec, FieldSpec, CodeBlock>> processFieldMet
           case GEO -> targetInterceptor = GeoField.class;
           case VECTOR -> targetInterceptor = VectorField.class;
         }
+      } else if (indexed != null && targetCls == null && isEnum(processingEnv, fieldType)) {
+        targetInterceptor = TextTagField.class;
```

Comment:
This raises an interesting question @bsbodden - are enums intended to be indexed as a tag or by their ordinal value? There are perfectly legitimate instances where they ought to be stored by their ordinal (e.g. bitwise flags)

Reply:
For my part redis om .NET allows for both - it depends on how the enum would be serialized by the JSON serializer, if it would be serialized to a number, then it's stored & queried as a numeric ordinal ( that's actually the default behavior), whereas if the serializer converts it to it's string representation, it is stored as text value and is indexed/queried as a tag.

Reply:
That's a good point. Currently the behavior is to serialize as a string and treat it as a TAG. But I can see a bitmask option being useful, my only issue would be how to deal with changes/addition to the Enum

Reply:
Spring Data Mongo seems to just do the same, Spring conversion. I'll add a issue to support Ordinal value, probably need a Gson annotation to attach a specific serializer for it.